articles/dms/tutorial-mysql-azure-mysql-offline-portal.md

```diff
@@ -0,0 +1,254 @@
+
+> [!IMPORTANT]
+> The “MySQL to Azure Database for MySQL” online migration scenario is being replaced with a parallelized, highly performant offline migration scenario from June 1, 2021. For online migrations, you can use this new offering together with [data-in replication](https://docs.microsoft.com/azure/mysql/concepts-data-in-replication). Alternatively, use open-source tools such as [MyDumper/MyLoader](https://centminmod.com/mydumper.html) with data-in replication for online migrations. 
+
+> [!INCLUDE [preview features callout](../../includes/dms-boilerplate-preview.md)]
+
+
+In this tutorial, you learn how to:
+
+> [!div class="checklist"]
+>
+> * Migrate database schema using mysqldump utility.
+> * Create an instance of Azure Database Migration Service.
+> * Create a migration project by using Azure Database Migration Service.
+> * Run the migration.
+> * Monitor the migration.
+
+## Prerequisites
+
+To complete this tutorial, you need to:
+
+* Have an Azure account with an active subscription. [Create an account for free] (https://azure.microsoft.com/free/?WT.mc_id=A261C142F).
+* Have an on-premises MySQL database with version 5.7 or 8. If not, then download and install [MySQL community edition](https://dev.mysql.com/downloads/mysql/) 5.7 or 8.
+* [Create an instance in Azure Database for MySQL](../mysql/quickstart-create-mysql-server-database-using-azure-portal.md). Refer to the article [Use MySQL Workbench to connect and query data](../mysql/connect-workbench.md) for details about how to connect and create a database using the Workbench application. The on-premises MySQL version must match with Azure Database for MySQL version. For example, MySQL 5.7 can only migrate to Azure Database for MySQL 5.7 and not upgraded to 8. 
+* Create a Microsoft Azure Virtual Network for Azure Database Migration Service by using Azure Resource Manager deployment model, which provides site-to-site connectivity to your on-premises source servers by using either [ExpressRoute](../expressroute/expressroute-introduction.md) or [VPN](../vpn-gateway/vpn-gateway-about-vpngateways.md). For more information about creating a virtual network, see the [Virtual Network Documentation](../virtual-network/index.yml), and especially the quickstart articles with step-by-step details.
+
+    > [!NOTE]
+    > During virtual networkNet setup, if you use ExpressRoute with network peering to Microsoft, add the *Microsoft.Sql* service [endpoint](../virtual-network/virtual-network-service-endpoints-overview.md) to the subnet in which the service will be provisioned. This configuration is necessary because Azure Database Migration Service lacks internet connectivity.
```

Comment:
Our target endpoint is MySQL so we need to add that to subnet.
We should use this from the online migration tutorial:

"During virtual network setup, if you use ExpressRoute with network peering to Microsoft, add the following service endpoints to the subnet in which the service will be provisioned:

Target database endpoint (for example, SQL endpoint, Cosmos DB endpoint, and so on)
Storage endpoint
Service bus endpoint
This configuration is necessary because Azure Database Migration Service lacks internet connectivity."

Reply:
Offline migration did not require Storage or Service Bus endpoint. I only had to add the Microsoft.Sql endpoint to the vnet that I created with peering? Is this specific to usage with ExpressRoute or for Online migration?

Reply:
All scenarios in DMS require storage and service bus endpoints, as that is how the resource provider (i.e. control plane) communicates with the data plane running the scenario.  It would be good to update the docs to mention all the required endpoints, or at least link to the concepts or other page in the DMS docs that describes all the required endpoints.